File: RadioControl/app/src/main/java/com/nikhilparanjape/radiocontrol/receivers/ServiceStarter.java
package com.nikhilparanjape.radiocontrol.receivers;

/**
 * Created by admin on 4/23/2017.
 */
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.os.Build;

import com.nikhilparanjape.radiocontrol.services.BackgroundAirplaneService;

public class ServiceStarter extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            context.startForegroundService(new Intent(context, BackgroundAirplaneService.class));
        } else {
            context.startService(new Intent(context, BackgroundAirplaneService.class));
        }
    }
}

File: RadioControl/app/src/main/java/com/nikhilparanjape/radiocontrol/receivers/ServiceStarter.kt
package com.nikhilparanjape.radiocontrol.receivers

/**
 * Created by admin on 4/23/2017.
 */

import android.annotation.SuppressLint
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.os.Build
import com.nikhilparanjape.radiocontrol.services.BackgroundJobService

class ServiceStarter : BroadcastReceiver() {

    @SuppressLint("UnsafeProtectedBroadcastReceiver") // Google bad
    override fun onReceive(context: Context, intent: Intent) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            context.startForegroundService(Intent(context, BackgroundJobService::class.java))
        } else {
            context.startService(Intent(context, BackgroundJobService::class.java))
        }
    }
}